handlers.go

```diff
@@ -3153,6 +3155,66 @@ func (s *Server) handleAddContentsToCollection(c echo.Context, u *User) error {
 	return c.JSON(200, map[string]string{})
 }
 
+// handleCommitCollection godoc
+// @Summary      Produce a CID of the collection contents
+// @Description  This endpoint is used to save the contents in a collection, producing a top-level CID that references all the current CIDs in the collection.
+// @Param        coluuid     path     string  true     "coluuid"
+// @Tags         collections
+// @Produce      json
+// @Success      200  {object}  string
+// @Router       /collections/commit/{collection-id} [post]
+func (s *Server) handleCommitCollection(c echo.Context, u *User) error {
+	colid := c.Param("coluuid")
+
+	var col Collection
+	if err := s.DB.First(&col, "uuid = ? and user_id = ?", colid, u.ID).Error; err != nil {
+		return err
+	}
+
+	contents := []Content{}
+	if err := s.DB.Debug().
+		Model(CollectionRef{}).
+		Where("collection = ?", col.ID).
+		Joins("left join contents on contents.id = collection_refs.content").
+		Select("contents.*").
+		Scan(&contents).Error; err != nil {
+		return err
+	}
+
+	bserv := blockservice.New(s.Node.Blockstore, offline.Exchange(s.Node.Blockstore))
+	dserv := merkledag.NewDAGService(bserv)
+
+	collectionNode := merkledag.ProtoNode{}
+	for _, content := range contents {
+		curCID := content.Cid.CID
+		nd, err := dserv.Get(context.Background(), curCID)
+		if err != nil {
+			return err
+		}
+
+		lnk, err := format.MakeLink(nd)
+		if err != nil {
+			return err
+		}
+		collectionNode.AddRawLink("a link", lnk)
+	}
+
+	// update DB with new collection CID
+	col.CID = collectionNode.Cid().String()
+	if err := s.DB.Save(&col).Error; err != nil {
```

Comment:
Avoid using `save` for updating (it is good for creating new rows), as it could update fields you don't intend to update. Instead do something like;

```
if err := s.DB.Model(Collection{}).Where("id = ?", col.ID).UpdateColumn("c_id", collectionNode.Cid().String()).Error; err != nil {
		return err
	}

Reply:
I did not know `save` did that!  There are other places where this occurs and I think I'd be good to change them too